internal/pkg/cli/svc_init.go

```diff
@@ -180,23 +182,19 @@ func newInitSvcOpts(vars initSvcVars) (*initSvcOpts, error) {
 	return opts, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error for any invalid optional flags.
 func (o *initSvcOpts) Validate() error {
-	if o.appName == "" {
-		return errNoAppInWorkspace
-	}
-	if o.wkldType != "" {
-		if err := validateSvcType(o.wkldType); err != nil {
-			return err
+	if o.appName != "" {
+		// This command must be run within the app's workspace.
+		if o.appName != o.wsAppName {
```

Comment:
If `tryReadingAppName()` can't find an app name in `.workspace`, this value will be `""` so if the user passes in an app flag outside of a workspace (which is when they would most likely pass an app flag), they will get this error about the workspace already being registered with "app [blank]" rather than `errNoAppInWorkspace`.

Maybe we should check for an empty `o.wsAppName` sooner?

Reply:
ooh good call